iothub_client/inc/iothub_device_client.h

```diff
@@ -124,8 +124,8 @@ extern "C"
     * @param    userContextCallback             User specified context that will be provided to the
     *                                           callback. This can be @c NULL.
     *
-    *           @b NOTE: The application behavior is undefined if the user calls
-    *           the IoTHubDeviceClient_Destroy() function from within any callback.
+    * @warning: Do not call IoTHubDeviceClient_Destroy() from inside your application's callback.
```

Comment:
Should this warning be also added to the `IoTHubDeviceClient_Destroy` API itself?

Reply:
Yup - adding